proto/api.go

```diff
@@ -236,7 +236,10 @@ func (sr *ScanResponse) Verify(req Request) error {
 //   representing the constituent requests.
 func (br *BatchRequest) Add(args Request) {
 	union := RequestUnion{}
-	union.SetValue(args)
+	if !union.SetValue(args) {
+		// TODO(tschottdorf) evaluate whether this should return an error.
+		log.Fatalf("batch request not in RequestUnion: %+v", args)
```

Comment:
A batch request is not the only invalid request to add to a union. I think this would be better as:

```
log.Fatalf("unable to add %T to batch request", args)
```